src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -162,11 +164,13 @@ public static RequestDelegateResult Create(MethodInfo methodInfo, Func<HttpConte
     private static FactoryContext CreateFactoryContext(RequestDelegateFactoryOptions? options) =>
         new()
         {
+            ServiceProvider = options?.ServiceProvider,
             ServiceProviderIsService = options?.ServiceProvider?.GetService<IServiceProviderIsService>(),
             RouteParameters = options?.RouteParameterNames?.ToList(),
             ThrowOnBadRequest = options?.ThrowOnBadRequest ?? false,
             DisableInferredFromBody = options?.DisableInferBodyFromParameters ?? false,
-            Filters = options?.RouteHandlerFilterFactories?.ToList()
+            Filters = options?.RouteHandlerFilterFactories?.ToList(),
+            EndpointMetadata = options?.DefaultEndpointMetadata
```

Comment:
```suggestion
            Metadata = new(options?.DefaultEndpointMetadata)
```

Then we don't need to remember the difference between `factoryContext.Metadata` and `factoryContext.EndpointMetadata`.